worldedit-core/src/main/java/com/sk89q/worldedit/internal/cui/ServerCUIHandler.java

```diff
@@ -128,24 +132,19 @@ public static BaseBlock createStructureBlock(Player player) {
         double xz = Math.cos(Math.toRadians(rotY));
         int x = (int) (location.getX() - (-xz * Math.sin(Math.toRadians(rotX))) * 12);
         int z = (int) (location.getZ() - (xz * Math.cos(Math.toRadians(rotX))) * 12);
-        int y = Math.max(0, Math.min(Math.min(255, posY + 32), posY + 3));
+        int y = Math.max(
+                player.getWorld().getMinY(),
+                Math.min(Math.min(player.getWorld().getMaxY(), posY + MAX_DISTANCE), posY + 3)
+        );
 
         CompoundBinaryTag.Builder structureTag = CompoundBinaryTag.builder();
 
         posX -= x;
         posY -= y;
         posZ -= z;
 
-        if (dataVersion >= 2526) {
-            if (Math.abs(posX) > 48 || Math.abs(posY) > 48 || Math.abs(posZ) > 48) {
-                // 20w16a+ allows structure blocks up to 48 per axis
-                return null;
-            } else {
-                if (Math.abs(posX) > 32 || Math.abs(posY) > 32 || Math.abs(posZ) > 32) {
-                    // Structure blocks on versions <= 20w16a have a limit of 32x32x32
-                    return null;
-                }
-            }
+        if (Math.abs(posX) > MAX_DISTANCE || Math.abs(posY) > MAX_DISTANCE || Math.abs(posZ) > MAX_DISTANCE) {
+            return null;
```

Comment:
We don't set max distance to anything other than 32

Reply:
Upstream merge for sake of not annotating it.

Reply:
Seems like a mistake...